src/openms_gui/source/VISUAL/TOPPViewIdentificationViewBehavior.cpp

```diff
@@ -330,112 +330,121 @@ namespace OpenMS
             {
               // if no fragment annotations are stored, create a theoretical spectrum
               addTheoreticalSpectrumLayer_(ph);
+
+              // synchronize PeptideHits with the annotations in the spectrum
+              current_layer.synchronizePeakAnnotations();
+              // remove labels and theoretical spectrum (will be recreated using PH annotations)
+              removeGraphicalPeakAnnotations_(spectrum_index);
```

Comment:
are these changes compatible with user added annotations?

Reply:
this is inside the `if (ph.getPeakAnnotations().empty())` code block, so it will be run only once per PeptideHit using the theoretical spectrum, when there are no PeakAnnotations there yet,
and before the user can add any annotations manually.
Immediately after that, the behaviour should be identical to reading in an ID file with PeakAnnotations.

Reply:
I experimented around with user annotations. After they are created from the theoretical spectrum, they are treated as peak annotations that were read in from an ID file.
They will be saved when saving a file, together with any user added annotations (or excluding any annotations the user deleted). Deleting all annotations in a spectrum will trigger the automatic annotation again, when the ID is reselected.

Reply:
Ok so this means that user annotations are preserved if spectra are changed

Reply:
Yes,
that line `removeGraphicalPeakAnnotations_(spectrum_index)` is only applied to the labels generated by the theoretical spectrum. A second set is recreated afterwards from the PeptideHit annotations, so if I don't delete them here, we would end up with two sets of the same labels.
It is a bit of unnecessary computing: creating labels, making annotations from these labels, deleting labels, recreating them from annotations.
But this way we can use the existing code for annotated IDs and this is only done once per PeptideHit.